agent/engine/task_manager.go

```diff
@@ -312,6 +312,15 @@ func (mtask *managedTask) handleContainerChange(containerChange dockerContainerC
 		}
 	}
 
+	// If the essential container is stopped, set the ExecutionStoppedAt timestamp
+	if container.GetKnownStatus() == api.ContainerStopped && container.IsEssential() {
+		now := mtask.time().Now()
+		ok := mtask.Task.SetExecutionStoppedAt(now)
+		if ok {
+			seelog.Infof("Essential container in task stopped, task %s, time: %s", mtask.Task.String(), now)
```

Comment:
Also, let's be more clear here that we're recording the execution stopped time: `"Recording execution stopped time for a task, essential container in task stopped"`